### PR TITLE
Some minor source code clean up tasks & refactoring

### PR DESCRIPTION
**Pull request by Jarmos-san**
## Description

As described in #21 the project was initially started but soon abandoned for whatever reasons. There's been a common consensus to get it back on track. As such the aforementioned issue thread tracks what needs to be taken care before the project can be restarted.

P.S: The contents of this PR is pretty huge for obvious reasons, so please bear with it for just this one time. 😅 🙏 

On that note, this PR contains a major clean up task which is only of everything that's in plan. To summarize, this PR has;

- Removed redundant `main.js`, `index.html`, `styles.css`, the `components` & `styles` directory which were otherwise cluttering the root directory. Some more random files & directories were also removed to avoid further clutter when starting out.
- Renamed the project in `package.json` (although note sure if it's necessary at all).
- Added `.editorconfig` & `.gitattributes` files for better cross-platform development practices.
- Convert arbitrary JavaScript files which existed before into proper TypeScript React (files with `.tsx` extensions) for later.

## Changes Include

- [x] Refactor ⚒️ (code change that neither fixes a bug nor adds a feature)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by Jarmos-san** [comment]
@varshney-himanshu there's a PR #30 waiting to be reviewed & merged. To avoid possible merge conflicts, I think we should review & merge this current PR & continue with the rest of the contributions. 🙏

**Review by varshney-himanshu — CHANGES_REQUESTED** [review]
@Jarmos-san I reviewed your branch - there seems to be an error with the import of the stylesheet that you have deleted. can you please resolve this before we can merge?

image

**Comment by Bhupesh-V** [comment]
@all-contributors please add @Jarmos-san for code, ideas,  maintenance, mentoring, review and projectManagement

**Pull request merged**
